source/docs/software/basic-programming/joystick.rst

```diff
@@ -0,0 +1,151 @@
+
+``XboxController`` Class
+------------------------
+
+.. image:: images/joystick/xbox.jpg
+
+.. tabs::
+
+   .. code-tab:: java
+
+      XboxController exampleXbox = new XboxController(0); // 0 is the USB Port to be used as indicated on the Driver Station
+
+   .. code-tab:: c++
+
+      XboxController exampleXbox{0}; // 0 is the USB Port to be used as indicated on the Driver Station
+
+The ``XboxController`` class provides named indicies for each of the buttons that you can access with ``XboxController.Button.kX.value``.  The rumble feature of the controller can be controlled by using ``XboxController.setRumble(GenericHID.RumbleType.kRightRumble, value)``.  Many users do a split stick arcade drive that uses the left stick for just forwards / backwards and the right stick for left / right turning.
+
+POV
+---
+
+.. image:: images/joystick/dpadangles.png
+   :alt: The angles used by the code of the POV/D-pad with 0 at the top and continuing clockwise.
+
+On joysticks the POV is directional hat that can select one of 8 different angles or read -1 for unpressed.  The XboxController D-pad works the same as a POV.  Be careful when using a POV with exact angle requirements as it is hard for the user to ensure they select exactly the angle desired.
+
+``GenericHID`` Usage
+--------------------
+
+An axis can be used with ``.getRawAxis(0)`` (if not using either of the classes above) that returns the current value.  Zero in this example is the index of axis as found in the Drive Station mentioned above.
+
+.. tabs::
+
+   .. code-tab:: java
+
+      private final PWMSparkMax m_leftMotor = new PWMSparkMax(0);
+      private final PWMSparkMax m_rightMotor = new PWMSparkMax(1);
+      private final DifferentialDrive m_robotDrive = new DifferentialDrive(m_leftMotor, m_rightMotor);
+      private final GenericHID m_stick = new GenericHID(0);
+
+      m_robotDrive.arcadeDrive(-m_stick.getY(), m_stick.getX());
+
+   .. code-tab:: c++
+
+      frc::PWMVictorSPX m_leftMotor{0};
+      frc::PWMVictorSPX m_rightMotor{1};
+      frc::DifferentialDrive m_robotDrive{m_leftMotor, m_rightMotor};
+      frc::GenericHID m_stick{0};
+
+      m_robotDrive.ArcadeDrive(-m_stick.GetY(), m_stick.GetX());
+
+Button Usage
+------------
+
+Unlike an axis you will usually want to use the ``pressed`` and ``released`` methods to respond to button input.  These will return true if the button has been activated since the last check.  This is helpful for taking an action once when the event occurs but not having to continuously do it while the button is held down.
+
+.. tabs::
+
+   .. code-tab:: java
+
+      if (joystick.getRawButtonPressed(0)) {
+         turnIntakeOn(); // While held intake stays on
```

Comment:
```suggestion
         turnIntakeOn(); // When pressed intake stays on
```

Reply:
You don't think it is important to distinguish here that because of the next line it will be turned off when the button is released.  I guess I could be talked into "When pressed the intake turns on" that would be an accurate description of just that line.

Reply:
right, your wording is better